pkg/skuba/actions/node/join/join.go

```diff
@@ -62,6 +62,7 @@ func Join(joinConfiguration deployments.JoinConfiguration, target *deployments.T
 	}
 
 	statesToApply := []string{
+		"kubeadm.reset",
```

Comment:
just to document. I would like to see this state run dynamically on failure to clean up. :)

Reply:
Seconded. We should always clean up after failure when possible.